packages/cli/src/commands/pg/credentials/url.ts

```diff
@@ -0,0 +1,70 @@
+import color from '@heroku-cli/color'
+import {Command, flags} from '@heroku-cli/command'
+import {Args, ux} from '@oclif/core'
+import {legacyEssentialPlan} from '../../../lib/pg/util'
+import {getAddon} from '../../../lib/pg/fetcher'
+import pgHost from '../../../lib/pg/host'
+import {URL} from 'url'
+import type {CredentialsInfo} from '../../../lib/pg/types'
+import heredoc from 'tsheredoc'
+
+export default class Url extends Command {
+  static topic = 'pg';
+  static description = 'show information on a database credential';
+  static flags = {
+    name: flags.string({
+      char: 'n',
+      description: 'which credential to show (default credentials if not specified)',
+      default: 'default',
+    }),
+    app: flags.app({required: true}),
+  };
+
+  static args = {
+    database: Args.string(),
+  };
+
+  public async run(): Promise<void> {
+    const {flags, args} = await this.parse(Url)
+    const {app, name} = flags
+    const {database} = args
+    const db = await getAddon(this.heroku, app, database)
+    if (legacyEssentialPlan(db) && name !== 'default') {
+      throw new Error('Legacy Essential-tier databases do not support named credentials.')
```

Comment:
I don't know if we want consistency in this, but I'm always replacing

```js
throw new Error(...)
```

with

```js
ux.error(...)
```

Reply:
Oh yeah, that's a good idea.